consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/MergeBlockProcessor.java

```diff
@@ -0,0 +1,289 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.consensus.merge;
+
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.ethereum.chain.MutableBlockchain;
+import org.hyperledger.besu.ethereum.core.Account;
+import org.hyperledger.besu.ethereum.core.Block;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.GoQuorumPrivacyParameters;
+import org.hyperledger.besu.ethereum.core.MutableWorldState;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.core.WorldUpdater;
+import org.hyperledger.besu.ethereum.mainnet.BlockProcessor;
+import org.hyperledger.besu.ethereum.mainnet.MainnetBlockProcessor;
+import org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor;
+import org.hyperledger.besu.ethereum.mainnet.MiningBeneficiaryCalculator;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Stream;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+
+public class MergeBlockProcessor extends MainnetBlockProcessor {
+  private static final Logger LOG = LogManager.getLogger();
+  private final MergeContext mergeContext;
+
+  public MergeBlockProcessor(
+      final MainnetTransactionProcessor transactionProcessor,
+      final TransactionReceiptFactory transactionReceiptFactory,
+      final Wei blockReward,
+      final MiningBeneficiaryCalculator miningBeneficiaryCalculator,
+      final boolean skipZeroBlockRewards,
+      final Optional<GoQuorumPrivacyParameters> goQuorumPrivacyParameters) {
+    super(
+        transactionProcessor,
+        transactionReceiptFactory,
+        blockReward,
+        miningBeneficiaryCalculator,
+        skipZeroBlockRewards,
+        goQuorumPrivacyParameters);
+    this.mergeContext = MergeContextFactory.get();
+  }
+
+  @Override
+  protected boolean rewardCoinbase(
+      final MutableWorldState worldState,
+      final BlockHeader header,
+      final List<BlockHeader> ommers,
+      final boolean skipZeroBlockRewards) {
+
+    if (!mergeContext.isPostMerge(header)) {
+      super.rewardCoinbase(worldState, header, ommers, skipZeroBlockRewards);
```

Comment:
the return of rewardCoinbase is not needed in this case ?

Reply:
Right, post-merge the block rewards will come from consensus chain

Reply:
But it's in this case it's before post merge `!mergeContext.isPostMerge(header)` or I miss something